### PR TITLE
Add Teams Integration and converted forecast to dollars

### PR DESCRIPTION
**Pull request by subashc2023**
1) Upgraded Python Lambda function
2) Added Slack Integration and Teams integration to Lambda Function
3) Improved instructions in Readme to create a Function, Procedure, and scheduling it to run on a cron job
4) Added Secret handling to CF Template.
5) Added a TOTAL Snowflake Usage row.

### CODE REVIEW

**Review by jimzucker — DISMISSED** [review]
LGTM - @subash please merge

**Review by jimzucker — DISMISSED** [review]
LGTM - Please merge

File: snowflake-usage-monitor-cf.yaml [hunk 1]
```diff
@@ -144,56 +144,81 @@ Resources:
...
+              except Exception as err:
+                  status_code = 400
+
+              return {"statusCode": status_code}
       Description: "Echo Lambda created from CloudFormation template"
       FunctionName: !Ref "lambdaName"
       Handler: "index.handler"
       Role: !GetAtt "lambdaExecutionIAMRole.Arn" 
-      Runtime: "python3.9"
+      Runtime: "python3.7"
```
Comment (jimzucker) [inline snowflake-usage-monitor-cf.yaml]: is this python version correct?

File: index.py [hunk 2]
```diff
@@ -1,48 +1,78 @@
...
+            database, forecast, change = row[1:]
+            if not database.endswith("(GB)"):
+                forecast = f"${forecast}"
+            else:
+                forecast = str(forecast)
+            str_forecast = str(forecast).ljust(forecast_width)
+            result.append(f"{database.ljust(account_width)} | {str_forecast} | {change.strip()}")
+        
+        secret = get_secret(SECRET_MANAGER, REGION)
+        result.append("* Forecast components may be ±15%")
```
Comment (jimzucker) [inline index.py]: this looks wrong

**Review by jimzucker — CHANGES_REQUESTED** [review]

File: snowflake-usage-monitor-cf.yaml (hunk 1 above)
Comment (subashc2023) [reply snowflake-usage-monitor-cf.yaml]: yes. requests library is deprecated in 3.9, and so i downgraded to 3.7

File: index.py (hunk 2 above)
Comment (subashc2023) [reply index.py]: This has been pushed

**Comment by subashc2023** [comment]
image

**Review by jimzucker — CHANGES_REQUESTED** [review]
Readme looks out of date, not update with current output in $$ and also does not explain the calcs for $$

File: METERING_HISTORY_TREND.sql [hunk 3]
```diff
@@ -17,6 +17,15 @@ specific language governing permissions and limitations
 under the License.
 */
 
+/*
+    There are 2 select statements that convert USAGE_IN_CREDITS to USAGE_IN_CURRENCY. 
+    They do this by multiplying the USAGE_IN_CREDITS by the COST PER CREDIT MULTIPLER. 
+    This MULTIPLIER depends on the type of account that you have.
+    For a BASIC account, it costs $2 per credit.
+    For an ENTERPRISE Account, it cosst $3.7 per credit.
+    If you have an Enterprise Account, please change the value of "multipler" in the first line of code from 2 to 3.7
```
Comment (jimzucker) [inline METERING_HISTORY_TREND.sql]: @subashc2023 we have to make this either
a. determine itself by running some queries(preferred)
b. make an option they pick when they run the installation

I am guessing there is a query we can run to figure out ifew are Enterprise or not

**Review by jimzucker — CHANGES_REQUESTED** [review]

File: METERING_HISTORY_TREND.sql (hunk 3 above)
Comment (subashc2023) [reply METERING_HISTORY_TREND.sql]: This has been automated in the latest push.

**Comment by subashc2023** [comment]
image

**Review by jimzucker — APPROVED** [review]
LGTM Please merge it

**Pull request merged**